### PR TITLE
Move basis change to main compute function

### LINKED ISSUE #992: Reduce amount of repeated code in `_compute` by converting bases `xyz` to/from `rpz` in the `_compute` method
First raised in #924 , there is lots of duplicate code to just call `rpz2xyz_vec`, so we could probably just move that into the recursive `_compute` method and check for vector dims and use the basis argument there. Might need to add a `native_basis` kwarg to the compute fun decorator to note the few items that are natively computed in `xyz` (like the `SplineXYZ` or `FourierXYZ` curves)

### LINKED ISSUE #1088: Quantities are computed wrong unless returned data is in cylindrical basis vectors
We only want to do a basis conversion on quantities that directly compute a vector's contravariant components in the cylindrical coordinate system. Otherwise we'll be converting basis vectors from cylindrical to Cartesian multiple times because the dependencies are already in cartesian.

### PR DESCRIPTION
**Pull request by YigitElma**
Resolves #992
Resolves #1088

All compute quantities are now returned in toroidal $(R,\phi,Z)$ coordinates.  

TODO:

- [x] Implement `_curve` computes too
- [x] Add `test_compute_everthing_xyz`

### CODE REVIEW

**Comment by YigitElma** [comment]
So, the point of this PR is to get rid of repeated code. But if I do so, the test `test_data_index_deps` fails because we don't have `data["phi"]` inside compute functions anymore. I can try to add `phi` in that test. But I want to see all other tests pass first.

File: desc/compute/utils.py [hunk 1]
```diff
@@ -140,6 +140,16 @@ def _compute(
         data = data_index[parameterization][name]["fun"](
             params=params, transforms=transforms, profiles=profiles, data=data, **kwargs
         )
+        # to reduce repeated code, for 3D and naturally in rtz but desired in xyz basis
+        # quantities can be converted to xyz basis here
+        if (
```
Comment (dpanici) [inline desc/compute/utils.py:145]: move this outside of recursive, have _compute functions assume things come in in rpz

**Comment by dpanici** [comment]
Enforce passed-in data is in rpz always (add to docstring)

File: tests/test_data_index.py [hunk 2]
```diff
@@ -111,6 +111,13 @@ def test_data_index_deps(self):
...
                 # assert correct dependencies are queried
-                assert queried_deps[p][name]["data"] == data | axis_limit_data, err_msg
+                # TODO: conversion from rpz to xyz is taken out of actual function
+                #       registration because of this data["phi"] is not queried in
+                #       the source code but actually needed for the computation. This
+                #       is a temporary fix until we have a better way to automatically
+                #       handle this.
+                assert queried_deps[p][name]["data"].issubset(
+                    data | axis_limit_data
+                ), err_msg
```
Comment (unalmis) [inline tests/test_data_index.py:121]: If you can't find a better way, then let's instead subtract phi from the later set. FYI we removed the tests that ensures everything can be computed individually, i.e. without computing the entire dependency tree, and replaced it with this test which explicitly checks for this via inspecting the source code. For functions like the third derivatives of the basis vectors it's useful to keep this check as strict as possible.

Comment (YigitElma) [reply tests/test_data_index.py:121]: Substracting "phi" may cause problems since we only want to remove it if the quantity is one of the 3d vectors which changed basis in the main compute function. 
This was a temporary change. i will think of a better way to do it, once I finish the actual coding part.

File: desc/compute/utils.py [hunk 3]
```diff
@@ -141,6 +141,19 @@ def _compute(
...
         )
 
+    # to reduce repeated code, 3D quantities can be converted to xyz basis here.
+    # Assumes every 3D quantity is in rpz basis.
+    for name in names:
+        if (
+            data_index[parameterization][name]["dim"] == 3  # it should be 3D
+            and kwargs.get("basis", "rpz") == "xyz"  # user should ask in xyz
+            and ("phi" in data)  # phi is needed for conversion
+            and data_index[parameterization][name]["coordinates"] == "rtz"
```
Comment (dpanici) [inline desc/compute/utils.py:151]: post which tests fail when this is removed

**Comment by YigitElma** [comment]
Make every compute function to return in rpz by default and make all of the basis changes in main compute function

**Comment by YigitElma** [comment]
Apply basis conversion to _curve compute functions too

**Comment by YigitElma** [comment]
Can someone who uses coils implement the same thing for _curve compute functions? @dpanici @ddudt @kianorr I couldn't figure out what caused it to fail and this is probably an easy thing for you guys. I don't know which parameter should be in which basis by default.

File: desc/compute/_surface.py [hunk 4]
```diff
@@ -47,13 +46,10 @@ def _x_FourierRZToroidalSurface(params, transforms, profiles, data, **kwargs):
...
     data=["x"],
     parameterization="desc.geometry.core.Surface",
-    basis="{'rpz', 'xyz'}: Basis for returned vectors, Default 'rpz'",
 )
 def _X_Surface(params, transforms, profiles, data, **kwargs):
     coords = data["x"]
-    if kwargs.get("basis", "rpz").lower() == "rpz":
-        # if basis is rpz, then "x" is rpz and we must convert to xyz
-        coords = rpz2xyz(coords)
+    coords = rpz2xyz(coords)
```
Comment (ddudt) [inline desc/compute/_surface.py:52]: Why are we converting this to Cartesian coordinates? I thought the whole point of this PR is to return everything in $(R,\phi,Z)$ coordinates?

File: desc/compute/_surface.py [hunk 5]
```diff
@@ -71,13 +67,10 @@ def _X_Surface(params, transforms, profiles, data, **kwargs):
...
     data=["x"],
     parameterization="desc.geometry.core.Surface",
-    basis="{'rpz', 'xyz'}: Basis for returned vectors, Default 'rpz'",
 )
 def _Y_Surface(params, transforms, profiles, data, **kwargs):
     coords = data["x"]
-    if kwargs.get("basis", "rpz").lower() == "rpz":
-        # if basis is rpz, then "x" is rpz and we must convert to xyz
-        coords = rpz2xyz(coords)
+    coords = rpz2xyz(coords)
```
Comment (ddudt) [inline desc/compute/_surface.py:73]: see previous comment

File: desc/compute/_surface.py (hunk 4 above)
Comment (YigitElma) [reply desc/compute/_surface.py:52]: I thought it wouldn't make sense that `X` returns `R`, or `Y` returns `Phi`. We already have `R`, `Phi` and `Z` for `rpz` coordinates.

Comment (YigitElma) [reply desc/compute/_surface.py:52]: X, Y, R and Phi should be the only data type that have strict basis definition, if I remember correctly

Comment (ddudt) [reply desc/compute/_surface.py:52]: Oh I see now. Yes this makes sense

**Comment by ddudt** [comment]
> Can someone who uses coils implement the same thing for _curve compute functions? @dpanici @ddudt @kianorr I couldn't figure out what caused it to fail and this is probably an easy thing for you guys. I don't know which parameter should be in which basis by default.

I will work on this

File: desc/compute/utils.py [hunk 6]
```diff
@@ -91,6 +92,27 @@ def compute(parameterization, names, params, transforms, profiles, data=None, **
...
+
+            if name == "x":
+                data[name] = rpz2xyz(data[name])
+            else:
+                errorif(
+                    "phi" not in data.keys(),
+                    ValueError,
+                    "'phi' must be included in the compute data "
+                    + "to convert to 'xyz' basis.",
+                )
```
Comment (ddudt) [inline desc/compute/utils.py:113]: It would be convenient if we could automatically add `"phi"` to the list of compute quantity `names` when `basis="xyz"` is requested. Unfortunately I think that logic would have to be added to each class's `compute` method, since that is typically where we get the params that are needed for the list of compute quantities.

**Comment by ddudt** [comment]
> > Can someone who uses coils implement the same thing for _curve compute functions? @dpanici @ddudt @kianorr I couldn't figure out what caused it to fail and this is probably an easy thing for you guys. I don't know which parameter should be in which basis by default.
> 
> I will work on this

@YigitElma this should be implemented for curves now

File: desc/compute/utils.py (hunk 6 above)
Comment (YigitElma) [reply desc/compute/utils.py:113]: In some curve functions, this was previously done by using transforms. So, I had a different if statement for `phi` is not included in the `data`. But yeah just having `phi` always should help.

**Comment by YigitElma** [comment]
> > > Can someone who uses coils implement the same thing for _curve compute functions? @dpanici @ddudt @kianorr I couldn't figure out what caused it to fail and this is probably an easy thing for you guys. I don't know which parameter should be in which basis by default.
> > 
> > 
> > I will work on this
> 
> @YigitElma this should be implemented for curves now

Thanks! Then only thing left for thisPR is the `test_compute_everything_xyz`

**Comment by YigitElma** [comment]
I don't understand how the new test fails. I created the new pickle file using the (commented) test and pushed that file to use it in the actual test. Maybe the precision of the devices github actions use is different? Or it could also be that I made a mistake in the test.

File: desc/compute/utils.py [hunk 7]
```diff
@@ -91,6 +95,35 @@ def compute(parameterization, names, params, transforms, profiles, data=None, **
         data=data,
         **kwargs,
     )
+
+    # convert data from default 'rpz' basis to 'xyz' basis, if requested by the user
+    parameterization = _parse_parameterization(parameterization)
+    for name in data.keys():
```
Comment (f0uriest) [inline desc/compute/utils.py:101]: can we swap the order of the loop and the `if basis == xyz` part? That way we won't have to check every qty if we're not doing any conversion

File: desc/compute/utils.py [hunk 8]
```diff
@@ -91,6 +95,35 @@ def compute(parameterization, names, params, transforms, profiles, data=None, **
...
+
+            if name == "x":
+                data[name] = rpz2xyz(data[name])
+            else:
+                if "phi" in data:
+                    data[name] = rpz2xyz_vec(data[name], phi=data["phi"])
+                else:
+                    # compute phi if it's not already in data. Adding phi
+                    # to data at the beginning cause matrix dimension problems
+                    # TODO: Maybe look into that later (why error?)
```
Comment (f0uriest) [inline desc/compute/utils.py:116]: what error were you getting? was it just a test failing or an actual error?

Comment (YigitElma) [reply desc/compute/utils.py:116]: It was a matrix dimension error. I didn't go too deep into this but when I append phi directly to names before recursive _compute function, dimension of phi was not proper for usage in rpz2xyz_vec function. I felt like appending it at the beginning cause it not to use given grid (that seemed the only reason for dimension difference), or something like that

File: tests/test_compute_funs.py [hunk 9]
```diff
@@ -1298,6 +1299,159 @@ def test_compute_everything():
     assert not error
 
 
+@pytest.mark.unit
+def test_compute_everything_xyz():
```
Comment (f0uriest) [inline tests/test_compute_funs.py:1303]: we could combine this with the previous one and use `pytest.mark.parametrize`

Would also be good to explicitly say `basis="rpz"` in the previous one.

File: tests/test_data_index.py (hunk 2 above)
Comment (f0uriest) [reply tests/test_data_index.py:121]: I think this should be ok? We're ensuring that the dependencies are a subset of what actually gets computed right? as opposed to ensuring equality?

File: desc/coils.py [hunk 10]
```diff
@@ -224,7 +224,7 @@ def compute_magnetic_field(
             current = params.pop("current", self.current)
 
         data = self.compute(
-            ["x", "x_s", "ds"], grid=source_grid, params=params, basis="xyz"
+            ["phi", "x", "x_s", "ds"], grid=source_grid, params=params, basis="xyz"
```
Comment (f0uriest) [inline desc/coils.py:227]: do we need to add phi here? is this because of the issue with adding it automatically in the compute func?

File: desc/compute/utils.py (hunk 8 above)
Comment (f0uriest) [reply desc/compute/utils.py:116]: can you post the error? I would think if you just do 
```
if basis == "xyz" and "phi" not in names:
    names += ["phi"]
```
should work fine?

Comment (YigitElma) [reply desc/compute/utils.py:116]: I am away from my laptop now, but I can try to send it around dev meeting

File: tests/test_compute_funs.py (hunk 9 above)
Comment (YigitElma) [reply tests/test_compute_funs.py:1303]: Yeah I also thought of that. I will combine them

File: desc/coils.py (hunk 10 above)
Comment (YigitElma) [reply desc/coils.py:227]: @ddudt added that one

File: desc/compute/utils.py (hunk 8 above)
Comment (ddudt) [reply desc/compute/utils.py:116]: I'm not certain this will always work, because the `params` and `transforms` were already set for the original list of names and might not contain the info needed to compute `"phi"`

File: desc/compute/_curve.py [hunk 11]
```diff
@@ -173,10 +159,10 @@ def _Z_Curve(params, transforms, profiles, data, **kwargs):
     coordinates="s",
     data=["s"],
     parameterization="desc.geometry.curve.FourierPlanarCurve",
-    basis="{'rpz', 'xyz'}: Basis for returned vectors, Default 'rpz'",
     basis_in="{'rpz', 'xyz'}: Basis for input params vectors, Default 'xyz'",
 )
 def _x_FourierPlanarCurve(params, transforms, profiles, data, **kwargs):
+    # convert to xyz for displacement and rotation
     if kwargs.get("basis_in", "xyz").lower() == "rpz":
```
[the remark this answers is not in the document]
Comment (ddudt) [reply desc/compute/_curve.py:166]: `"basis_in"` is specific to the `FourierPlanarCurve` class. It allows the curve parameters to be defined in either basis, and we decided this was better than creating a whole new Curve class (like `FourierPlanarCurveXYZ` vs `FourierPlanarCurveRPZ`). We still need this and it should be correct as is.

File: desc/coils.py (hunk 10 above)
Comment (ddudt) [reply desc/coils.py:227]: Yes I added this before we added the logic to automatically add phi when needed. I think this might still be necessary because computing `"phi"` can require additional `params` relative to the other compute quantities for some Curve classes. See my other comment about this

File: desc/compute/utils.py (hunk 8 above)
Comment (dpanici) [reply desc/compute/utils.py:116]: `"phi"` (currently, this will change with the general angle stuff) only needs `grid` to be included in `transforms` which I think is almost always the case for basically any compute quantity

File: desc/objectives/_geometry.py [hunk 12]
```diff
@@ -711,8 +711,8 @@ def build(self, use_jit=True, verbose=1):
                 params=self._surface.params_dict,
                 transforms=surface_transforms,
                 profiles={},
-                basis="xyz",
             )["x"]
+            surface_coords = rpz2xyz(surface_coords)
```
Comment (dpanici) [inline desc/objectives/_geometry.py:715]: why is this needed? can't you still request `basis=xyz`?

File: tests/test_curves.py [hunk 13]
```diff
@@ -55,19 +55,15 @@ def test_torsion(self):
     def test_frenet(self):
         """Test frenet-serret frame of circular curve."""
         c = FourierRZCurve()
-        data = c.compute(
-            ["frenet_tangent", "frenet_normal", "frenet_binormal"], basis="rpz", grid=0
-        )
+        data = c.compute(["frenet_tangent", "frenet_normal", "frenet_binormal"], grid=0)
```
Comment (dpanici) [inline tests/test_curves.py:58]: why remove the basis arguments in these tests?

File: tests/test_curves.py [hunk 14]
```diff
@@ -55,19 +55,15 @@ def test_torsion(self):
...
         np.testing.assert_allclose(T, np.array([[0, 1, 0]]), atol=1e-12)
         np.testing.assert_allclose(N, np.array([[-1, 0, 0]]), atol=1e-12)
         np.testing.assert_allclose(B, np.array([[0, 0, 1]]), atol=1e-12)
         c.rotate(angle=np.pi)
         c.flip([0, 1, 0])
         c.translate([1, 1, 1])
-        data = c.compute(
-            ["frenet_tangent", "frenet_normal", "frenet_binormal"], basis="xyz", grid=0
-        )
+        data = c.compute(["frenet_tangent", "frenet_normal", "frenet_binormal"], grid=0)
```
Comment (dpanici) [inline tests/test_curves.py:66]: same as above comment

**Comment by dpanici** [comment]
Make sure `basis` is removed from compute function decorators

File: desc/objectives/_geometry.py (hunk 12 above)
Comment (YigitElma) [reply desc/objectives/_geometry.py:715]: Inside the objectives, we don't use `compute` but instead use `_compute` which is jittable. Since the latter is recursive, I put conversion to the former. So, only for objective cases, you should change basis manually

File: tests/test_curves.py (hunk 13 above)
Comment (YigitElma) [reply tests/test_curves.py:58]: @ddudt

File: tests/test_curves.py (hunk 14 above)
Comment (YigitElma) [reply tests/test_curves.py:66]: @ddudt

File: desc/objectives/_geometry.py (hunk 12 above)
Comment (dpanici) [reply desc/objectives/_geometry.py:715]: Got it, thanks. Might be worth noting this in the [making an objective function tutorial in the docs](https://desc-docs.readthedocs.io/en/latest/adding_objectives.html) then

Comment (YigitElma) [reply desc/objectives/_geometry.py:715]: It is documented in the `compute` and `_compute` functions

File: tests/test_curves.py (hunk 13 above)
Comment (ddudt) [reply tests/test_curves.py:58]: "rpz" is now the default basis

File: tests/test_curves.py (hunk 14 above)
Comment (ddudt) [reply tests/test_curves.py]: @dpanici I reverted this change, but are you sure this test is supposed to use both bases? It's difficult for me to visualize the geometry that is being tested here, since there are flips/rotations/etc. Same for the other two Frenet tests in this file.

Comment (ddudt) [reply tests/test_curves.py]: FYI with this reversion the test is now failing, but I think that's an unrelated bug that I still need to track down.

File: desc/geometry/core.py [hunk 15]
```diff
@@ -441,14 +453,18 @@ def compute(
...
+        method = kwargs.pop("method", "auto")
         if params is None:
-            params = get_params(names, obj=self)
+            params = get_params(names, obj=self, basis=kwargs.get("basis", "rpz"))
         if transforms is None:
             transforms = get_transforms(
                 names,
                 obj=self,
                 grid=grid,
                 jitable=kwargs.pop("jitable", False),
```
Comment (ddudt) [inline desc/geometry/core.py:465]: Does anyone know why we use `False` as the default here for the `Surface` class? For the `Curve` class we hard-code this to `True`.

File: tests/test_curves.py (hunk 14 above)
Comment (ddudt) [reply tests/test_curves.py]: Actually this change is needed to make the test pass (both compute calls need to return "rpz" basis). I'm about 80% sure the old test was incorrect and it is fixed now, but whoever wrote this original test should check it.

File: desc/geometry/core.py (hunk 15 above)
Comment (f0uriest) [reply desc/geometry/core.py:465]: `jitable=True` means it uses the direct transforms, which for 1d curves is pretty cheap. For a 2d surface its a bit more expensive (especially if you need high resolution)

File: tests/test_curves.py (hunk 14 above)
Comment (f0uriest) [reply tests/test_curves.py]: yes they should both be in rpz. It's just testing the directions of the frenet frame before and after flip/rotate etc. It's not messing with any basis stuff.

File: desc/compute/utils.py [hunk 16]
```diff
@@ -140,11 +164,10 @@ def _compute(
         data = data_index[parameterization][name]["fun"](
             params=params, transforms=transforms, profiles=profiles, data=data, **kwargs
         )
-
     return data
 
 
-def get_data_deps(keys, obj, has_axis=False):
+def get_data_deps(keys, obj, has_axis=False, basis="rpz"):
```
Comment (ddudt) [inline desc/compute/utils.py:170]: For a future PR: The variable we call `keys` here and in the other util functions below is the same as what we call `names` in the `compute` functions above and in other files. We should probably rename for consistency across the code.

File: tests/test_curves.py (hunk 14 above)
Comment (ddudt) [reply tests/test_curves.py]: @f0uriest there are 3 different `test_frenet` functions in this file for different curve parameterizations:

1. `FourierRZCurve` now uses `basis="rpz"` for both computes
2. `FourierXYZCurve` uses `basis="rpz"` for the first compute and `basis="xyz"` for the second compute
3. `FourierPlanarCurve` uses `basis="xyz" for both computes

All three tests now pass, but should they really all be using "rpz" throughout?

Comment (f0uriest) [reply tests/test_curves.py]: Iirc when I made them I was using whatever the "native" basis was for the curve, but I don't know how much has changed since then

File: desc/compute/__init__.py [hunk 17]
```diff
@@ -63,10 +63,10 @@ def _build_data_index():
     for p in data_index:
         for key in data_index[p]:
             full = {
-                "data": get_data_deps(key, p, has_axis=False),
-                "transforms": get_derivs(key, p, has_axis=False),
-                "params": get_params(key, p, has_axis=False),
-                "profiles": get_profiles(key, p, has_axis=False),
+                "data": get_data_deps(key, p, has_axis=False, basis="rpz"),
```
[the remark this answers is not in the document]
Comment (f0uriest) [reply desc/compute/__init__.py:66]: the only thing different would be `phi` but i don't think we want to compute that for everything if its not needed (right now its free but in the future may require building additional transforms)

File: desc/compute/utils.py [hunk 18]
```diff
@@ -91,16 +96,37 @@ def compute(parameterization, names, params, transforms, profiles, data=None, **
...
+                    data[name] = rpz2xyz_vec(data[name], phi=data["phi"])
+
     return data
 
 
-# TODO: can we do the basis change here possibly? instead of repeating it everywhere?
-#  Maybe give an "inherent basis" parameter to the compute decorator
-#  so we check what the natural basis is versus what the desired is?
 def _compute(
     parameterization, names, params, transforms, profiles, data=None, **kwargs
```
Comment (f0uriest) [inline desc/compute/utils.py:120]: maybe add a check that basis isnt in kwargs here? (it gets popped in the other function so wouldn't normally be passed in here)

File: desc/magnetic_fields/_core.py [hunk 19]
```diff
@@ -1835,9 +1835,16 @@ def compute(
...
-            params = get_params(names, obj=self)
+            params = get_params(names, obj=self, basis=kwargs.get("basis", "rpz"))
         if transforms is None:
-            transforms = get_transforms(names, obj=self, grid=grid, **kwargs)
+            transforms = get_transforms(
+                names,
+                obj=self,
+                grid=grid,
+                basis=kwargs.get("basis", "rpz"),
+                method=kwargs.pop("method", "auto"),
```
Comment (f0uriest) [inline desc/magnetic_fields/_core.py:1845]: could this cause confusion with the spline method?

File: tests/test_compute_funs.py [hunk 20]
```diff
@@ -1257,45 +1258,77 @@ def test_compute_everything():
...
+    error_rpz = False
+    error_xyz = False
 
     for p in things:
-        this_branch_data[p] = things[p].compute(
-            list(data_index[p].keys()), **grid.get(p, {})
+        this_branch_data_rpz[p] = things[p].compute(
+            list(data_index[p].keys()), **grid.get(p, {}), basis="rpz"
+        )
+        this_branch_data_xyz[p] = things[p].compute(
```
Comment (f0uriest) [inline tests/test_compute_funs.py:1275]: think you need to remove the tensor qtys before computing here

File: desc/compute/__init__.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (ddudt) [reply desc/compute/__init__.py:66]: But if we don't include `"phi"` here then won't this potentially not work when trying to compute some thing in the xyz basis?

**Review by f0uriest — APPROVED** [review]

File: desc/compute/utils.py [hunk 21]
```diff
@@ -91,16 +96,38 @@ def compute(parameterization, names, params, transforms, profiles, data=None, **
...
+    # convert data from default 'rpz' basis to 'xyz' basis, if requested by the user
+    if basis == "xyz":
+        from .geom_utils import rpz2xyz, rpz2xyz_vec
+
+        for name in data.keys():
+            errorif(
+                data_index[p][name]["dim"] == (3, 3),
+                NotImplementedError,
+                "Tensor quantities cannot be converted to Cartesian coordinates.",
+            )
```
Comment (unalmis) [inline desc/compute/utils.py:109]: [Yes they can](https://github.com/PlasmaControl/DESC/issues/1110#issuecomment-2221187485)

**Review by ddudt — DISMISSED** [review]
We should wait to get an approval from at least one person who hasn't contributed to this PR.

File: desc/compute/utils.py [hunk 22]
```diff
@@ -248,8 +282,8 @@ def get_profiles(keys, obj, grid=None, has_axis=False, jitable=False, **kwargs):
         Grid to compute quantity on.
     has_axis : bool
         Whether the grid to compute on has a node on the magnetic axis.
-    jitable: bool
```
Comment (dpanici) [inline desc/compute/utils.py:251]: This was unused?

File: tests/test_curves.py [hunk 23]
```diff
@@ -66,7 +66,7 @@ def test_frenet(self):
         c.flip([0, 1, 0])
         c.translate([1, 1, 1])
         data = c.compute(
-            ["frenet_tangent", "frenet_normal", "frenet_binormal"], basis="xyz", grid=0
```
Comment (dpanici) [inline tests/test_curves.py:69]: Interesting that these vectors are the same in xyz and rpz

**Review by dpanici — APPROVED** [review]

File: tests/test_data_index.py (hunk 2 above)
Comment (YigitElma) [reply tests/test_data_index.py:123]: Actually since we deleted `phi` from dependencies, the previous version of this test might still be passing.

**Review by dpanici — CHANGES_REQUESTED** [review]
Just realized, we should update the change log to reflect this as it's pretty major (that by default the compute function will return things in rpz, as before I think it wasn't always the case?)

**Review by ddudt — APPROVED** [review]

**Review by dpanici — APPROVED** [review]

**Review by f0uriest — APPROVED** [review]

File: tests/test_data_index.py (hunk 2 above)
Comment (unalmis) [reply tests/test_data_index.py:123]: @YigitElma In that case can you change this back

Comment (unalmis) [reply tests/test_data_index.py:123]: Nevermind I changed it back on my branch.
